tests/results/SourceRangeTest.py

```diff
@@ -86,6 +86,18 @@ def test_json(self):
                                       end_line=4).__json__(use_relpath=True)
         self.assertEqual(uut['start'], self.result_fileB_line2)
 
+    def test_str(self):
+        uut = str(SourceRange.from_values('test_file', 1, 1, 2, 1))
+        compare = abspath('test_file') + ': L1 C1 : L2 C1'
+        self.assertEqual(uut, compare)
+
+        uut = str(SourceRange.from_values('test_file', 1, None, 2, None))
+        compare = abspath('test_file') + ': L1 : L2'
+        self.assertEqual(uut, compare)
+
+        uut = str(SourceRange.from_values('test_file', None, None, None, None))
+        self.assertEqual(uut, abspath('test_file'))
```

Comment:
But, calculating last line and last column would be a problem as you said - _for example_
if `start.line` is not `None` and `end.line` is `None`, then `filename: L3 : ?lastline?`
I'm not sure properly as to what to do when `end.line` or `end.start` is `None` :/